src/graph/entities/graph_entity.c

```diff
@@ -73,6 +73,26 @@ SIValue GraphEntity_Keys
 	return keys;
 }
 
+// returns an SIArray of all keys and values in graph entity properties. 
+// Keys at even positions, Values at odd position of the array
+SIValue GraphEntity_Properties
+(
+	const GraphEntity *e
+) {
+	GraphContext *gc = QueryCtx_GetGraphCtx();
+	const AttributeSet set = GraphEntity_GetAttributes(e);
+	int propCount = ATTRIBUTE_SET_COUNT(set);
+	SIValue properties = SI_Array(propCount);
```

Comment:
why this is not return map?

Reply:
You are right, it is better to return map. Change done in [2e22b7b](https://github.com/RedisGraph/RedisGraph/pull/2632/commits/2e22b7b92699fc7027530007f0047022fdca9018)